server/src/test/java/org/elasticsearch/discovery/StableMasterDisruptionIT.java

```diff
@@ -0,0 +1,169 @@
+        assertThat(internalCluster().getMasterName(), equalTo(masterNode));
+    }
+
+    /**
+     * Verify that nodes fault detection works after master (re) election
+     */
+    public void testFollowerCheckerDetectsUnresponsiveNodeAfterMasterReelection() throws Exception {
+        internalCluster().startNodes(4,
+            Settings.builder()
+                .put(LeaderChecker.LEADER_CHECK_TIMEOUT_SETTING.getKey(), "1s")
+                .put(LeaderChecker.LEADER_CHECK_RETRY_COUNT_SETTING.getKey(), "10")
+                .put(FollowersChecker.FOLLOWER_CHECK_TIMEOUT_SETTING.getKey(), "1s")
+                .put(FollowersChecker.FOLLOWER_CHECK_RETRY_COUNT_SETTING.getKey(), 1).build());
+        ensureStableCluster(4);
+
+        logger.info("--> stopping current master");
+        internalCluster().stopCurrentMasterNode();
+
+        ensureStableCluster(3);
+
+        final String master = internalCluster().getMasterName();
+        final List<String> nonMasters = Arrays.stream(internalCluster().getNodeNames()).filter(n -> master.equals(n) == false)
+            .collect(Collectors.toList());
+        final String isolatedNode = randomFrom(nonMasters);
+        final String otherNode = nonMasters.get(nonMasters.get(0).equals(isolatedNode) ? 1 : 0);
+
+        logger.info("--> isolating [{}]", isolatedNode);
+
+        final NetworkDisruption networkDisruption = new NetworkDisruption(new NetworkDisruption.TwoPartitions(
+            singleton(isolatedNode), Stream.of(master, otherNode).collect(Collectors.toSet())), new NetworkUnresponsive());
+        setDisruptionScheme(networkDisruption);
+        networkDisruption.startDisrupting();
+
+        logger.info("--> waiting for master to remove it");
+        ensureStableCluster(2, master);
+
+        networkDisruption.stopDisrupting();
+        ensureStableCluster(3);
+    }
+
+    /**
+     * Verify that nodes fault detection works after master (re) election
+     */
+    public void testFollowerCheckerDetectsDisconnectedNodeAfterMasterReelection() throws Exception {
+        internalCluster().startNodes(4);
+        ensureStableCluster(4);
+
+        logger.info("--> stopping current master");
+        internalCluster().stopCurrentMasterNode();
+
+        ensureStableCluster(3);
+
+        final String master = internalCluster().getMasterName();
+        final List<String> nonMasters = Arrays.stream(internalCluster().getNodeNames()).filter(n -> master.equals(n) == false)
+            .collect(Collectors.toList());
+        final String isolatedNode = randomFrom(nonMasters);
+        final String otherNode = nonMasters.get(nonMasters.get(0).equals(isolatedNode) ? 1 : 0);
+
+        logger.info("--> isolating [{}]", isolatedNode);
+
+        final NetworkDisruption networkDisruption = new NetworkDisruption(new NetworkDisruption.TwoPartitions(
+            singleton(isolatedNode), Stream.of(master, otherNode).collect(Collectors.toSet())), new NetworkDisconnect());
```

Comment:
just `Sets.newHashSet(master, otherNode)`

Reply:
Thanks, I knew there was such a thing but couldn't remember what it was called.